### PR TITLE
docs: add CLI reference

### PR DESCRIPTION
**Pull request by jpetey75**
Closes: https://github.com/lightdash/lightdash/issues/11224

### Description:

Adding CLI reference doc

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: docs/docs/references/lightdash-cli.mdx [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+
+This command will bring up an interactive list of projects in your organization to choose from, like this:
+
+![lightdash config set-project project selector](./assets/set-project.png)
+
+If you need to set the project non-interactively you can use one of the two optional arguments below.
+
+| Option   | Required?   | Description                                                                                      |
+| -------- | ----------- | ------------------------------------------------------------------------------------------------ |
+| `--name` | 🟢 Argument |  Set the project non-interactively by passing a project name                                            |
```
Comment (ZeRego) [inline docs/docs/references/lightdash-cli.mdx:127]: I found these tables & columns confusing. We have a column for 'option' and then in the 'required?' it shows 'Argument' or 'Option'. 

Should they be different tables? 

Arguments 

| Argument    | Required?  |  Description                                                                                      |
| --------- | ----------- | ------------------------------------------------------------------------------------------------ |
| `[URL]`   | ✅  | The URL for your Lightdash instance                                                               |

Options

| Option   | Description                                                                                      |
| ---------| ------------------------------------------------------------------------------------------------ |   
| `--token [string]`   | For logging in via access token - most common for SSO users - exclude to login with email/password|
| `--example [number]` |  example number option |
| `--skip-questions [boolean]` |  example boolean option |

**Comment by jpetey75** [comment]
@ZeRego updates are finished - I stopped using the tables for options in most of the later sections, do you like the simplified bullet layout? Or do you think the table like the `lightdash preview` section uses is better?

**Comment by ZeRego** [comment]
@jpetey75 Build process found a broken link 

```
1:03:21 AM: Exhaustive list of all broken links found:
1:03:21 AM: 
1:03:21 AM: - On source page path = /references/lightdash-cli:
1:03:21 AM:    -> linking to (#lightdash-config-set-project) (resolved as: /references/()
```

**Comment by ZeRego** [comment]
> @ZeRego updates are finished - I stopped using the tables for options in most of the later sections, do you like the simplified bullet layout? Or do you think the table like the `lightdash preview` section uses is better?

Yes, we can have all the options in a bullet layout. 👍

**Comment by jpetey75** [comment]
@ZeRego this is ready for review - the only section I have outstanding question is `lightdash generate`. The CLI help suggests that the supported dbt options are different for that command than others, is that true?

**Comment by ZeRego** [comment]
> @ZeRego this is ready for review - the only section I have outstanding question is `lightdash generate`. The CLI help suggests that the supported dbt options are different for that command than others, is that true?

The only command that supports all dbt options is the `lightdash dbt`. 
The rest of the Lightdash commands support a subset of options for the `dbt list` and `dbt compile` commands.
    
List found here: https://github.com/lightdash/lightdash/blob/3aa0d858d050f531cde3dd1e97605f780806d31c/packages/cli/src/handlers/dbt/compile.ts#L28-L42

**Review by ZeRego — APPROVED** [review]

**Pull request merged**